### ISSUE TITLE
Add Justified HorizontalAlignment for RichTextBox

### ISSUE THREAD

**Issue by aliandead** [dotnet/winforms#10792]
### Background and motivation

I noticed that RTF documents support Justified alignment, yet RichTextBoxes do not provide the ability to set that alignment for some reasons.

### API Proposal

```csharp
    public enum HorizontalAlignment
    {
        Left = 0,
        Right = 1,
        Center = 2,
        Justified = 3
    }
```

### API Usage

```csharp
RichTextBox richTextBox = new RichTextBox();
richTextBox.SelectionAlignment = HorizontalAlignment.Justified;
```

### Risks

None.

### Will this feature affect UI controls?

It should be supported by the designer. It won't have any impact on accessibility. And obviously it doesn't need to be localized.

**Comment by elachlan** [comment]
`HorizontalAlignment` isn't only used by `RichEdit`, its used by many other controls.

https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.horizontalalignment?view=windowsdesktop-8.0#remarks

> This enumeration is used in numerous classes. A partial list of these classes is [CheckedListBox](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.checkedlistbox?view=windowsdesktop-8.0), [ColumnHeader](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.columnheader?view=windowsdesktop-8.0), [ComboBox](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.combobox?view=windowsdesktop-8.0), [ControlPaint](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.controlpaint?view=windowsdesktop-8.0), [Label](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.label?view=windowsdesktop-8.0), [ListBox](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.listbox?view=windowsdesktop-8.0), [Control](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.control?view=windowsdesktop-8.0), [RichTextBox](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.richtextbox?view=windowsdesktop-8.0), and [TextBox](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.textbox?view=windowsdesktop-8.0).

**Comment by aliandead** [comment]
(Whops) In that case it would make more sense to create a new enumeration, something like "TextAlignment"

**Comment by elachlan** [comment]
I'd suggest `ParagraphAlignment` because it closely aligns with the docs.

@merriemcgaw this would be a breaking change. Is it unlikely to be accepted?

**Comment by merriemcgaw** [comment]
Yeah, I'm afraid if this is any sort of breaking change it would be unlikely to be accepted.  Let's add the untriaged label and it can get discussed with the broader team, but I'd keep expectations in line.

**Comment by SteveAndrews** [comment]
@merriemcgaw What about adding a new property with the new enumeration and `[Obsolete]` the old property?

**Comment by merriemcgaw** [comment]
@SteveAndrews we could create a second property to manage this, I don't know if we'd obsolete the original property though it sounds like we could make a non-breaking way to do this.

That said, I don't believe this will bubble up far enough priority-wise to make it to .NET 9. We can keep this on our radar for .NET 10 because properly done I can see how it would benefit richedit customers.
